packages/clay-drop-down/src/Menu.tsx

```diff
@@ -173,15 +171,58 @@ const ClayDropDownMenu = React.forwardRef<HTMLDivElement, IProps>(
 		// See https://github.com/microsoft/TypeScript/issues/30748#issuecomment-480197036
 		ref
 	) => {
-		const subPortalRef = React.useRef<HTMLDivElement | null>(null);
+		const subPortalRef = useRef<HTMLDivElement | null>(null);
 
-		useDropdownCloseInteractions(
-			[alignElementRef, subPortalRef],
-			onSetActive,
-			focusRefOnEsc
-		);
+		useEffect(() => {
+			const handleClick = (event: MouseEvent) => {
+				const nodeRefs = [alignElementRef, subPortalRef];
+				const nodes: Array<Node> = (Array.isArray(nodeRefs)
+					? nodeRefs
+					: [nodeRefs]
+				)
+					.filter((ref) => ref.current)
+					.map((ref) => ref.current!);
+
+				if (
+					event.target instanceof Node &&
+					!nodes.find((element) =>
+						element.contains(event.target as Node)
+					)
+				) {
+					onSetActive(false);
+				}
+			};
+
+			window.addEventListener('mousedown', handleClick);
+
+			return () => {
+				window.removeEventListener('mousedown', handleClick);
+			};
+		}, []);
+
+		useEffect(() => {
+			const handleEsc = (event: KeyboardEvent) => {
+				if (event.key === Keys.Esc) {
+					event.stopImmediatePropagation();
+
+					if (focusRefOnEsc && focusRefOnEsc.current) {
+						focusRefOnEsc.current.focus();
+					}
+
+					onSetActive(false);
+				}
+			};
+
+			if (active) {
+				document.addEventListener('keyup', handleEsc, true);
+			}
```

Comment:
This could potentially be added multiple times right? I think we need to also remove the listener for whenever `active` is false.

Reply:
Yeah, we are doing this on `return` after that. With dependency with `active`, we will ensure that it is removed when it is false, but it is not added.

Reply:
Oh I see, I thought the return only happened when the component was being unmounted and destroyed

Reply:
> I thought the return only happened when the component was being unmounted and destroyed

Yeah, it will only happen when we have an empty dependency on `useEffect` but as in this case I'm set `active`, it will call the previous cleaning up before calling the effect again.